### PR TITLE
Add vendoring section

### PR DESCRIPTION
**Pull request by strk**
Addresses #178

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -57,6 +57,17 @@ After running for a while, the command should print
 ```
 ALL TESTS PASSED
 ```
+## Vendoring
+
+We keep a cached copy of dependencies under the vendor/ directory,
+managing updates via the `govendor` command (github.com/kardianos/govendor)
```
Comment (bkcsoft) [inline CONTRIBUTING.md]: maybe change to `[govendor](https://github.com/kardianos/govendor)` instead to be consistent :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -57,6 +57,17 @@ After running for a while, the command should print
...
+
+Pull requests should only include vendor/ updates IFF they are
```
Comment (bkcsoft) [inline CONTRIBUTING.md]: s/IFF/IF/

Comment (bkcsoft) [reply CONTRIBUTING.md]: Also prefer **bold** over CAPS

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -57,6 +57,17 @@ After running for a while, the command should print
...
+part of the same change, be it a bugfix or a feature addition.
+
+In any case, the vendor/ change needs to be justified as part
```
Comment (bkcsoft) [inline CONTRIBUTING.md]: `The vendor update needs to be`

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (strk) [reply CONTRIBUTING.md]: IFF stands for `IF and only IF`

Comment (strk) [reply CONTRIBUTING.md]: Anyway, dropped IFF as of 15bd941 (the "only" part is already expressed before)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -57,6 +57,17 @@ After running for a while, the command should print
...
 ```
+## Vendoring
+
+We keep a cached copy of dependencies under the vendor/ directory,
+managing updates via [govendor](http://github.com/kardianos/govendor).
+
+Pull requests should only include vendor/ updates if they are
+part of the same change, be it a bugfix or a feature addition.
+
+In any case, the vendor/ update needs to be justified as part
```
Comment (bkcsoft) [inline CONTRIBUTING.md]: I meant get rid of `In any case,`, but that might've been ambiguous :slightly_smiling_face:

Comment (strk) [reply CONTRIBUTING.md]: Yes, it was. See how you like ac644fd

**Comment by andreynering** [comment]
LGTM

**Comment by lunny** [comment]
LGTM. And changes of `CONTRIBUTING.md` should be agreed with all the three owners. @tboerger

**Review by bkcsoft — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -57,6 +57,17 @@ After running for a while, the command should print
 ```
 ALL TESTS PASSED
 ```
+## Vendoring
+
+We keep a cached copy of dependencies under the vendor/ directory,
```
Comment (tboerger) [inline CONTRIBUTING.md]: within the vendor/

**Review by tboerger — CHANGES_REQUESTED** [review]
And I would put the vendor/ into backticks, makes it better visible as a folder name.

**Comment by tboerger** [comment]
You forgot the backticks around the vendor/ mentions

**Comment by strk** [comment]
I can't see any missing backtick in the files changed tab, can you comment there if you do ?

**Comment by strk** [comment]
https://github.com/go-gitea/gitea/pull/187/commits/39b3fcad1ddbef541affdc0a1b3e63606da0c85f

**Review by tboerger — APPROVED** [review]

**Comment by tboerger** [comment]
LGTM

**Comment by moltam** [comment]
The Vendoring section was removed in [this commit](https://github.com/go-gitea/gitea/commit/91d6c715ea3323cd4b06d21c1aae20364a47a2a0#diff-6a3371457528722a734f3c51d9238c13L60). Was this intentional?

**Comment by tboerger** [comment]
> The Vendoring section was removed in this commit. Was this intentional?

Not intentional, maybe somebody else got the time to add it back to all our repos?

**Comment by lunny** [comment]
fixed by https://github.com/go-gitea/gitea/pull/890, @moltam

**Pull request merged**